dipy/reconst/tests/test_cti.py

```diff
@@ -0,0 +1,195 @@
+are_perpendicular = np.isclose(dot_product, 0)
+bvecs2 = np.concatenate(([hsph_updated.vertices] * 2) +
+                        [hsph_updated90] + ([hsph_updated.vertices]))
+bvecs2 = np.append(bvecs2, [[0, 0, 0]], axis=0)
+bvals2 = np.array([0] * 20 + [1] * 20 + [1] * 20 + [0] * 20 + [0])
+# Creating the second gtab table:
+gtab2 = gradient_table(bvals2, bvecs2)
+# Defining Btens:
+e1 = bvecs1  # (81,3)
+e2 = bvecs2  # (81,3)
+e3 = np.cross(e1, e2)
+V = np.stack((e1, e2, e3), axis=-1)
+# transposing along 2nd and 3rd axis.
+V_transpose = np.transpose(V, axes=(0, 2, 1))
+B = np.zeros((81, 3, 3))  # initializing a btensor
+b = np.zeros((3, 3))
+for i in range(81):
+    b[0, 0] = bvals1[i]
+    b[1, 1] = bvals2[i]
+    B[i] = np.matmul(V[i], np.matmul(b, V_transpose[i]))
+
+# on providing btens, (bvals1,bvecs1) is ignored.
+gtab = gradient_table(bvals1, bvecs1, btens=B)
+S0 = 100
+# we've isotropic and anisotropic diffusion tensor distribution (DTD)
+anisotropic_DTD = _anisotropic_DTD()  # assuming these functions work correctly
+isotropic_DTD = _isotropic_DTD()
+
+DTDs = [
+    anisotropic_DTD,
+    isotropic_DTD,
+    np.concatenate((anisotropic_DTD, isotropic_DTD))
+]
+
+# label for each DTD, for the plot
+DTD_labels = ['Anisotropic DTD', 'Isotropic DTD', 'Combined DTD']
+
+for DTD in DTDs:
+    D = np.mean(DTD, axis=0)
+    C = qti.dtd_covariance(DTD)
+    params = np.concatenate(
+        (np.zeros((1, 1)),  # replaced np.log(1)[np.newaxis, np.newaxis] with 0
+            qti.from_3x3_to_6x1(D),
+            qti.from_6x6_to_21x1(C))).T
+    data = qti.qti_signal(gtab, D, C, S0=S0)[np.newaxis, :]
+
+    # use labels that describe the signal
+    plt.plot(data.ravel(), label=DTD_labels[i])
+
+plt.legend()
+plt.show()
+
+
+def construct_cti_params(evals, evecs, kt, fcvt):
+    fevals = evals.reshape((-1, evals.shape[-1])) #has shape: (1, 3)
+    fevecs = evecs.reshape((-1,) + evecs.shape[-2:])
+    fevecs = fevecs.reshape((1, -1)) #shape: (3, 3)
+    fkt = kt.reshape((-1, kt.shape[-1]))
+    # Concatenate all the flattened tensors
+    cti_params = np.concatenate((fevals.T, fevecs.T, fkt, fcvt), axis = 0)    
+    return np.squeeze(cti_params) #returns shape: (48, )
+
+
+def test_cti_prediction():
+    ctiM = cti.CorrelationTensorModel(gtab1, gtab2)
+    anisotropic_DTD = _anisotropic_DTD()
+    isotropic_DTD = _isotropic_DTD()
+
+    DTDs = [
+        anisotropic_DTD,
+        isotropic_DTD,
+        np.concatenate((anisotropic_DTD, isotropic_DTD))
+    ]
+
+    DTD_labels = ["anisotropic_DTD", "isotropic_DTD", "combined_DTD"]
+    for i, DTD in enumerate(DTDs):
+        D = np.mean(DTD, axis=0)
+        D_flat = np.squeeze(from_3x3_to_6x1(D)) #has shape:(6, )    #pretty useless, not needed
+        evals, evecs = decompose_tensor(D)  #evals:shape: (3, ) & evecs.shape: (3, 3)
+        C = qti.dtd_covariance(DTD)
+        C = qti.from_6x6_to_21x1(C)
+
+        #changing up cti : 
+        const = np.sqrt(2) 
```

Comment:
After looking to the maths with @kerkelae. The correct conversion between the CTI's and QTI's covariance tensors are: 

const = np.sqrt(2)
Ccti = np.zeros((21,1))
Ccti[0] = C[0]
Ccti[1] = C[1] 
Ccti[2] = C[2]
Ccti[3] = C[3] / const 
Ccti[4] = C[4] / const 
Ccti[5] = C[5] / const 
Ccti[6] = C[6] / 2
Ccti[7] = C[7] / 2
Ccti[8] = C[8] / 2
Ccti[9] = C[9] / 2
Ccti[10] = C[10] / 2
Ccti[11] = C[11] / 2
Ccti[12] = C[12] / 2
Ccti[13] = C[13] / 2
Ccti[14] = C[14] / 2
Ccti[15] = C[15] / 2 
Ccti[16] = C[16] / 2 
Ccti[17] = C[17] / 2 
Ccti[18] = C[18] / (2 * const)
Ccti[19] = C[19] / (2 * const)
Ccti[20] = C[20] / (2 * const)

Perhaps it cool be useful to have this conversion implemented in a function.